### PR TITLE
[SMTChecker] Fix ICE on ABI functions with no arguments

### LINKED ISSUE #11161: [Z3] Failed assertion for test with abi.encode()
## Description

```
pragma experimental SMTChecker;
contract C {
  function f() public { abi.encode(); }
}
```

throws

```
solc: /usr/local/include/z3++.h:1835: z3::expr z3::cast_ast<z3::expr>::operator()(z3::context &, Z3_ast): Assertion `Z3_get_ast_kind(c, a) == Z3_NUMERAL_AST || Z3_get_ast_kind(c, a) == Z3_APP_AST || Z3_get_ast_kind(c, a) == Z3_QUANTIFIER_AST || Z3_get_ast_kind(c, a) == Z3_VAR_AST' failed.
Aborted (core dumped)
```

during compilation `solc test.sol`

Not sure if we should report this upstream instead.

### PR DESCRIPTION
**Pull request by blishko**
Fixes #11161.

### CODE REVIEW

File: test/libsolidity/smtCheckerTests/abi/abi_encode_no_arguments.sol [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+contract C {
+	function f() pure public { 
+		bytes memory res = abi.encode();
+		assert(res.length == 0); // should hold
+		assert(res.length > 0); // should fail
+		res = abi.encodePacked();
+		assert(res.length == 0); // should hold
+		assert(res.length > 0); // should fail
+		abi.encodeWithSelector(0);
+		abi.encodeWithSignature("");
```
Comment (blishko) [inline test/libsolidity/smtCheckerTests/abi/abi_encode_no_arguments.sol]: Can we assert something about the return value of these? If not, I will remove them.

Comment (axic) [reply test/libsolidity/smtCheckerTests/abi/abi_encode_no_arguments.sol]: The first one should be `hex"00000000"`. The second should be `keccak256("")[0:4]`. Not sure the SMTEncoder can devise all these values though.

Comment (axic) [reply test/libsolidity/smtCheckerTests/abi/abi_encode_no_arguments.sol]: But their length should be 4 bytes both in both cases, which the SMTEncoder should be able to handle.

Comment (leonardoalt) [reply test/libsolidity/smtCheckerTests/abi/abi_encode_no_arguments.sol]: I don't think it is right now.

Comment (axic) [reply test/libsolidity/smtCheckerTests/abi/abi_encode_no_arguments.sol]: The only thing it is capable of telling that two calls with the same inputs should equal in output (as is able to track the variables leading up to the call), correct?

Comment (leonardoalt) [reply test/libsolidity/smtCheckerTests/abi/abi_encode_no_arguments.sol]: Exactly

Comment (axic) [reply test/libsolidity/smtCheckerTests/abi/abi_encode_no_arguments.sol]: I guess we can leave a comment and a failing assertion here (`assert(res.length == 4) // should hold (if we implement support)`), so if behaviour changes it is detected.

Comment (blishko) [reply test/libsolidity/smtCheckerTests/abi/abi_encode_no_arguments.sol]: That is exactly what I am about to push :)

**Review by leonardoalt — APPROVED** [review]

**Pull request merged**